tests/queries/0_stateless/01018_Distributed__shard_num.sql

```diff
@@ -3,6 +3,9 @@
 -- make the order static
 SET max_threads = 1;
 
+-- data should be inserted into Distributed table synchronously
+SET insert_distributed_sync = 1;
```

Comment:
I don't see how this should help, since there is only one INSERT into Distributed table, and this table contains only localhost shard, which is synchronous by default (due to prefer_localhost_replica=1).

What was the original problem?

Reply:
> What was the original problem?

Your test is unstable.

Reply:
There is no `prefer_localhost_replica` in this test.
Without specifying this setting explicitly, it is randomized.

Reply:
Yep, then this patch should helps (I've also looked at on report with this test failed to verify this).